src/code/z_lights.c

```diff
@@ -329,8 +329,7 @@ Lights* Lights_NewAndDraw(GraphicsContext* gfxCtx, u8 ambientR, u8 ambientG, u8
     Lights* lights;
     s32 i;
 
-    // TODO allocation should be a macro
-    lights = (Lights*)((int)gfxCtx->polyOpa.d - sizeof(Lights));
+    lights = GRAPH_ALLOC(gfxCtx, sizeof(Lights));
     gfxCtx->polyOpa.d = (void*)lights;
```

Comment:
```suggestion
```
Should be able to remove this as it is a part of the macro.

Reply:
This creates warnings. Same for other examples.

Reply:
Out of curiosity, could you post the warnings?